Project/Source/Importers/ModelImporter.cpp

```diff
@@ -461,8 +460,8 @@ static void ImportNode(const char* modelFilePath, JsonValue jMeta, const aiScene
 			aiMesh* assimpMesh = assimpScene->mMeshes[node->mMeshes[i]];
 
 			ComponentMeshRenderer* meshRenderer = gameObject->CreateComponent<ComponentMeshRenderer>();
-			meshRenderer->meshId = ImportMesh(modelFilePath, jMeta, assimpMesh, resourceIndex, bones);
-			meshRenderer->materialId = materialIds[assimpMesh->mMaterialIndex];
+			meshRenderer->SetMeshID(ImportMesh(modelFilePath, jMeta, assimpMesh, resourceIndex, bones));
+			meshRenderer->SetMaterialID(materialIds[assimpMesh->mMaterialIndex]);
```

Comment:
Shouldn't that be using the ChangeMaterial instead of SetMaterialID? If not, where are you increasing the reference?

Reply:
This is on purpose. The reference gets increased when initializing the component.

Reply:
OK, ty